plugins/nextcloud-rule-exclusions-before.conf

```diff
@@ -739,15 +792,38 @@ SecRule REQUEST_FILENAME "@endsWith /login" \
             ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:pass1-text,\
             ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:pass2"
 
-# Logout token
-SecRule REQUEST_FILENAME "@endsWith /logout" \
-    "id:9508420,\
+# Password reset used in newer version of Nextcloud
+SecRule REQUEST_FILENAME "@rx /lostpassword/set/[^/]+/[^/]+$" \
+    "id:9508411,\
     phase:1,\
     pass,\
     t:none,\
     nolog,\
     ver:'nextcloud-rule-exclusions-plugin/1.0.0',\
-    ctl:ruleRemoveTargetById=941120;ARGS:requesttoken"
+    ctl:ruleRemoveTargetByTag=OWASP_CRS;ARGS:json.password"
+
+# requesttoken is used when logging in or when authenticating via a public share
+# Matches:
+# /login
+# /s/share-id/authenticate/showShare
+# /login/v2/grant
+# /logout
+SecRule REQUEST_FILENAME "@rx /(?:login(?:/v[0-9\.]+/grant)?|s/[^/]+/authenticate/showShare|logout)$" \
```

Comment:
This rule is complex. It should have tests.

Reply:
I don't have or know all the payloads for all of the rules triggered, so there won't be complete coverage.

Reply:
No issue. But I think you should at least test that the regular expression matches the paths you listed in the comment.